### DISCUSSION TITLE
🛠️ Setting Up Rsbuild for Node.js & TypeScript without a Web Server 🚫🌐

### DISCUSSION THREAD

**Discussion by pavelbe** [web-infra-dev/rsbuild#1384] [Q&A]
Hello, Rsbuild Team! 👋

I'm exploring ways to use Rsbuild for a Node.js project with TypeScript that operates in the console, without launching a web server. I aim to create a Node.js application template with TypeScript using Rsbuild, but I'm facing difficulties in configuring tsconfig.json and rsbuild.config.ts.

Most information about Rsbuild seems to focus on web development, and I'm unsure how to adapt it for Node.js and TypeScript in a non-web server environment. I would greatly appreciate your assistance with:

1. 📝 Recommendations for setting up tsconfig.json for a Node.js project using Rsbuild.
2. 🔧 Examples of configuring rsbuild.config.ts to compile TypeScript code for Node.js without starting a web server.
3. 📚 If available, any ready-made project examples or templates that I could use as a starting point.

Thank you for your help and looking forward to your advice! 🙏

**Comment by chenjiahan** [comment]
Hi, I have just added an example project for Node.js & TypeScript, see: https://github.com/web-infra-dev/rsbuild/tree/main/examples/node

This example contains some basic setups. If you have any other questions, feel free to ask here. 😄

**Reply by pavelbe** [reply]
Thank you very much!👍

**Comment by pavelbe** [comment]
Thank you so much for providing the example! 👍

I have one more request: I need a script (a command in package.json, like "dev") that would allow me to work in development mode so that when I make changes to the code, I can immediately see those changes in the console.

For example, if I write console.log('Hello, World!');, save the file, I want that to show up immediately in the console. And if I change this to console.log('Hello, World! ❤️ 🔥 🚀!');, I want the changes to also appear instantly in the console, without having to restart the command. Can you help me set up a development mode like this, where changes are applied automatically?

https://github.com/web-infra-dev/rsbuild/assets/9510939/f4750e9a-3b8a-4938-bc72-4cad9f5de899

**Reply by chenjiahan** [reply]
you can use `rsbuild build --watch`: https://rsbuild.dev/guide/basic/cli#rsbuild-build

**Comment by pavelbe** [comment]
It doesn't work as expected 🤷‍♂️.
If I type "pnpm rsbuild build --watch", it runs, but it doesn't output to the console, instead in the console

start   Compiling...
● Server ━━━━━━━━━━━━━━━━━━━━━━━━━ (100%) done                                                                                
ready   Server compiled in 0.03 s

https://github.com/web-infra-dev/rsbuild/assets/9510939/b5b6852d-fccb-4937-a755-b069f53890a7

**Comment by chenjiahan** [comment]
The `rsbuild build' command only builds the source code, it does not execute the output code. 😂

**Reply by pavelbe** [reply]
I thought that "rsbuild build --watch" would watch for changes and execute immediately.

How can I make the dev..... command build and immediately execute the code in the console? In the src folder (not the folder in dist/build)

**Reply by chenjiahan** [reply]
try this: https://github.com/privatenumber/tsx
